File: Java/src/No1446.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

public class No1446 {
    public int maxPower(String s) {
        int ans = 0, count = 1;
        char prev = s.charAt(0);
        char[] cs = s.toCharArray();
        for (int i = 1; i < cs.length; ++i) {
            if (cs[i] != prev) {
                ans = Math.max(count, ans);
                prev = cs[i];
                count = 1;
            } else {
                count++;
            }
        }
        return ans;
    }
}

File: CPP/No1446.cc
/**
 * Created by Xiaozhong on 2021/12/1.
 * Copyright (c) 2021/12/1 Xiaozhong. All rights reserved.
 */

#include "iostream"
#include "vector"
#include "unordered_set"

using namespace std;

class Solution {
public:
    int maxPower(string s) {
        char prev = s[0];
        int cnt = 1;
        int idx = 1;
        int ans = 0;
        while (idx < s.size()) {
            if (prev == s[idx]) cnt++;
            else {
                ans = max(ans, cnt);
                cnt = 0;
                prev = s[idx];
            }
            idx++;
        }
        return max(ans, cnt);
    }
};

int main() {
    Solution s;
    cout << s.maxPower("leetcode") << endl;
}
